packages/ai-terminal/src/browser/ai-terminal-contribution.ts

```diff
@@ -0,0 +1,194 @@
+    ) {
+        this.chatContainer = document.createElement('div');
+        this.chatContainer.className = 'ai-terminal-chat-container';
+
+        const chatCloseButton = document.createElement('span');
+        chatCloseButton.className = 'closeButton codicon codicon-close';
+        chatCloseButton.onclick = () => this.dispose();
+        this.chatContainer.appendChild(chatCloseButton);
+
+        const chatResultContainer = document.createElement('div');
+        chatResultContainer.className = 'ai-terminal-chat-result';
+        this.chatResultParagraph = document.createElement('p');
+        this.chatResultParagraph.textContent = 'How can I help you?';
+        chatResultContainer.appendChild(this.chatResultParagraph);
+        this.chatContainer.appendChild(chatResultContainer);
+
+        this.chatInputContainer = document.createElement('div');
+        this.chatInputContainer.className = 'ai-terminal-chat-input-container';
+
+        this.chatInput = document.createElement('textarea');
+        this.chatInput.className = 'theia-input theia-ChatInput';
+        this.chatInput.placeholder = 'Ask about a terminal command...';
+        this.chatInput.onkeydown = event => {
+            if (event.key === 'Enter' && !event.shiftKey) {
+                event.preventDefault();
+                if (!this.haveResult) {
+                    this.send();
+                } else {
+                    this.terminalWidget.sendText(this.chatResultParagraph.innerText);
+                    this.dispose();
+                }
+            } else if (event.key === 'Escape') {
+                this.dispose();
+            } else if (event.key === 'ArrowUp' && this.haveResult) {
+                this.updateChatResult(this.getNextCommandIndex(1));
+            } else if (event.key === 'ArrowDown' && this.haveResult) {
+                this.updateChatResult(this.getNextCommandIndex(-1));
+            }
+        };
+        this.chatInputContainer.appendChild(this.chatInput);
+
+        const chatInputOptionsContainer = document.createElement('div');
+        const chatInputOptionsSpan = document.createElement('span');
+        chatInputOptionsSpan.className = 'codicon codicon-send option';
+        chatInputOptionsSpan.title = 'Send';
+        chatInputOptionsSpan.onclick = () => this.send();
+        chatInputOptionsContainer.appendChild(chatInputOptionsSpan);
+        this.chatInputContainer.appendChild(chatInputOptionsContainer);
+
+        this.chatContainer.appendChild(this.chatInputContainer);
+
+        terminalWidget.node.appendChild(this.chatContainer);
+
+        this.chatInput.focus();
+    }
+
+    protected async send(): Promise<void> {
+        const userRequest = this.chatInput.value;
+        if (userRequest) {
+            this.chatInput.value = '';
+
+            this.chatResultParagraph.innerText = 'Loading';
+            this.chatResultParagraph.className = 'loading';
+
+            const cwd = (await this.terminalWidget.cwd).toString();
```

Comment:
This will return an URI. Is a file uri (as opposed to a FS path) what the terminal expects?

Reply:
Good question. The return value is just piped through to the LLM, so it does not really matter I guess